pandas-stubs/core/indexes/multi.pyi

```diff
@@ -27,31 +29,42 @@ from pandas._typing import (
 class MultiIndex(Index[Any]):
     def __new__(
         cls,
-        levels=...,
-        codes=...,
-        sortorder=...,
+        levels: SequenceNotStr[SequenceNotStr[Hashable] | AnyArrayLike] = ...,
+        codes: SequenceNotStr[SequenceNotStr[int]] = ...,
+        sortorder: int | None = ...,
         names: SequenceNotStr[Hashable] = ...,
-        dtype=...,
-        copy=...,
+        copy: bool = ...,
         name: SequenceNotStr[Hashable] = ...,
         verify_integrity: bool = ...,
         _set_identity: bool = ...,
     ) -> Self: ...
     @classmethod
     def from_arrays(
-        cls, arrays, sortorder=..., names: SequenceNotStr[Hashable] = ...
+        cls,
+        arrays: SequenceNotStr[SequenceNotStr[Hashable] | AnyArrayLike],
+        sortorder: int | None = ...,
```

Comment:
This one's a bit odd

The name is `from_arrays`, and the pandas docstring says that this should be a sequence of array-like

But, the pandas docs (and tests) have several examples of passing lists, which are not `ArrayLike`

`lists` are not included in `AnyArrayLike`, so I've used `SequenceNotStr[Hashable] | AnyArrayLike`

Reply:
I think it should be `Sequence[Axes]` because this works:
```python
>>> pd.MultiIndex.from_arrays([range(2), [3,4]], names=["a", "b"])
MultiIndex([(0, 3),
            (1, 4)],
           names=['a', 'b'])
>>> pd.MultiIndex.from_arrays([{1:"x", 2:"y"}, [3,4]], names=["a", "b"])
MultiIndex([(1, 3),
            (2, 4)],
           names=['a', 'b'])
```

Reply:
> 🤔 maybe `ListLike` would be more suitable

Problem is that `ListLike` allows a single string, so we can't use it.  I think `Axes` is the right choice.